src/Core__String.resi

```diff
@@ -0,0 +1,1017 @@
+on MDN.
+*/
+@send
+external toLocaleUpperCase: string => string = "toLocaleUpperCase"
+
+/**
+`trim(str)` returns a string that is `str` with whitespace stripped from both
+ends. Internal whitespace is not removed.
+See [`String.trim`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim)
+on MDN.
+
+## Examples
+
+```rescript
+String.trim("   abc def   ") == "abc def"
+String.trim("\n\r\t abc def \n\n\t\r ") == "abc def"
+```
+*/
+@send
+external trim: string => string = "trim"
+
+/**
+`trimStart(str)` returns a string that is `str` with whitespace stripped from
+the beginning of a string. Internal whitespace is not removed.
+See [`String.trimStart`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimStart)
+on MDN.
+
+## Examples
+
+```rescript
+String.trimStart("   Hello world!   ") == "Hello world!   "
+String.trimStart("   Hello   world!   ") == "Hello   world!   "
+```
+*/
+@send
+external trimStart: string => string = "trimStart"
+
+/**
+`trinEnd(str)` returns a string that is `str` with whitespace stripped from the
+end of a string. Internal whitespace is not removed.
+See [`String.trimEnd`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimEnd)
+on MDN.
+
+## Examples
+
+```rescript
+String.trimEnd("   Hello world!   ") == "   Hello world!"
+String.trimEnd("   Hello   world!   ") == "   Hello   world!"
+```
+*/
+@send
+external trimEnd: string => string = "trimEnd"
+
+/**
+`padStart(str, n, padStr)` returns a string that has been padded with `padStr`
+(multiple times, if needed) until the resulting string reaches the given `n`
+length. The padding is applied from the start of the current string.
+See [`String.padStart`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart)
+on MDN.
+
+## Examples
+
+```rescript
+String.padStart("abc", 5, " ") == "  abc"
+String.padStart("abc", 6, "123465") == "123abc"
+```
+*/
+@send
+external padStart: (string, int, string) => string = "padStart"
+
+/**
+`padEnd(str, n, padStr)` returns a string that has been padded with `padStr`
+(multiple times, if needed) until the resulting string reaches the given `n`
+length. The padding is applied from the end of the current string.
+See [`String.padEnd`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padEnd)
+on MDN.
+
+## Examples
+
+```rescript
+String.padEnd("Hello", 10, ".") == "Hello....."
+String.padEnd("abc", 1, "") == "abc"
+```
+*/
+@send
+external padEnd: (string, int, string) => string = "padEnd"
+
+// TODO: add docs
+@get_index external getSymbol: (string, Core__Symbol.t) => option<'a> = ""
+@get_index external getSymbolUnsafe: (string, Core__Symbol.t) => 'a = ""
+@set_index external setSymbol: (string, Core__Symbol.t, 'a) => unit = ""
```

Comment:
It remains to document these three.

Reply:
👍

Reply:
I've never used `Symbol` in JS. Is there any example case for these bindings?

```res
String.getSymbol("name", Symbol.make("name"))
```

Compile to:
```js
"name"[Symbol("name")]
```

Reply:
Yeah I don't know, maybe symbols in strings isn't a thing and this binding is unnecessary?

I say we can skip docstrings for this now, and add a separate issue to investigate if these are to be removed.

Reply:
See https://github.com/rescript-association/rescript-core/issues/80